### ISSUE TITLE
Add color field to front/rear ports

### ISSUE THREAD

**Issue by williamwu0220** [netbox-community/netbox#3682]
### Environment
* Python version:  3.7.3
* NetBox version:  2.6.7

### Proposed Functionality
Using color keystones makes data center management more convenient, just like this:
https://imgur.com/a/P8izFXW
Since cables can be assigned color attributes in NetBox, I think it would be a big plus if front ports have this attributes as well. :)

### Use Case
For example, in our data center, we use yellow keystones denoting they're used by ip phone system, while green keystones means they're used by wireless system.

### Database Changes
After digging into source code, I think we can add ColorField in FrontPort class.

### External Dependencies
No.

**Comment by kobayashi** [comment]
I am curious about this feature though, do you have any idea to add this?
Is it like add color column next to "Connection" in front ports list?
![Screen Shot 2019-11-12 at 9 10 56 AM](https://user-images.githubusercontent.com/8712070/68678980-0132e000-052d-11ea-802d-b0ed49e8f532.png)

**Comment by williamwu0220** [comment]
TL;DR:
======

Yes, I think it would be good if we can see each color column directly in front ports list, just like cable list does:

![image](https://user-images.githubusercontent.com/41606/68735134-b0e47c80-0617-11ea-8daf-10d15686afdb.png)

Further more, It would be even more better if we can see each front panel color when tracing a cable linking:

![image](https://user-images.githubusercontent.com/41606/68735410-7f1fe580-0618-11ea-9411-44c638ef1a78.png)

Detailed Description
===================

Let me describe how we tracing cable linking in our environment. When an equipment needs connect to internet, we check the steps below:

1. Findout the location id of telecommunications outlet and its associated code via floor plan:
![image](https://user-images.githubusercontent.com/41606/68738686-e9895380-0621-11ea-874d-8146f1ba48e2.png)
or do on-site confirming:
![image](https://user-images.githubusercontent.com/41606/68738785-29e8d180-0622-11ea-929f-93374cac0e4e.png)

2. Go to data center finding out corresponding patch panel and keystone, double checking if color-code matched:
![image](https://user-images.githubusercontent.com/41606/68739027-cf9c4080-0622-11ea-9dc1-d3ff98aaec14.png)

3. Finally, according manual mantained mapping table, we can trace the far end interface port:
![image](https://user-images.githubusercontent.com/41606/68739198-3cafd600-0623-11ea-8b1a-2ee1ea19e30a.png)

So, it's really helpful if front ports have color columns. ;)

**Comment by williamwu0220** [comment]
Hi,
Do I need provide more detail usage scenarios? I'm waiting this feature for adopting netbox in our data center. :)

**Comment by williamwu0220** [comment]
Hi,
I'm willing to pick up this task, but I need to know if this feature request is acceptable. ;)

**Comment by kobayashi** [comment]
Thank you @williamwu0220 
This issue is accepted.

**Comment by jeremystretch** [comment]
@williamwu0220 have you made any progress on this? Should I remove you as the assigned owner?

**Comment by eachmobi** [comment]
Sorry, busy on works these days, I will work on this issue. :)

**Comment by williamwu0220** [comment]
Sorry, busy for works these days, I'll work on this issue soon :)
